app/assets/src/components/views/discovery/DiscoverySidebar.jsx

```diff
@@ -0,0 +1,223 @@
+
+    this.state = {
+      stats: {
+        samples: 0,
+        projects: 0
+        // TODO (gdingle):
+        // avg_reads_per_sample: 0,
+      },
+      metadata: {
+        host: {},
+        tissue: {},
+        created_at: {}
+        // TODO (gdingle):
+        // location: {},
+      },
+      genomes: {}
+    };
+  }
+
+  async componentDidMount() {
+    // TODO (gdingle): include genome name in payload from getSamples
+    const genomes = keyBy("id", await getAllHostGenomes());
+    this.setState({ genomes });
+  }
+
+  static getDerivedStateFromProps(newProps, prevState) {
+    const { currentTab, projects } = newProps;
+
+    if (currentTab == "samples") {
+      const samples = DiscoverySidebar.processSamples(
+        newProps.samples,
+        prevState.genomes
+      );
+      if (!samples || !samples.length) {
+        return prevState;
+      }
+      return {
+        stats: {
+          samples: samples.length,
+          projects: uniqBy("project_id", samples).length
+          // TODO (gdingle): reads not in samples data yet
+          // avg_reads_per_sample: 0,
+        },
+        metadata: {
+          host: countBy("host_genome", samples),
+          tissue: countBy("sample_tissue", samples),
+          created_at: countBy("created_at", samples)
+          // TODO (gdingle): location not in samples data yet
+          // location: {},
+        },
+        _computed: currentTab
+      };
+    } else if (currentTab == "projects") {
+      if (!projects || !projects.length) {
+        return prevState;
+      }
+
+      const hosts = flatten(map("hosts", projects));
+      const tissues = flatten(map("tissues", projects));
+
+      const created_ats = map(
+        p => DiscoverySidebar.formatDate(p.created_at),
+        projects
+      );
+
+      return {
+        z: {
+          samples: sumBy("number_of_samples", projects),
+          projects: projects.length
+          // TODO (gdingle): reads not in projects data yet
+          // avg_reads_per_sample: 0,
+        },
+        metadata: {
+          // TODO (gdingle): these freq counts per project, not per sample
+          host: countBy(_ => _, hosts),
+          tissue: countBy(_ => _, tissues),
+          created_at: countBy(_ => _, created_ats)
+          // TODO (gdingle): location not in projects data yet
+          // location: {},
+        },
+        _computed: currentTab
+      };
+    } else {
+      // eslint-disable-next-line no-console
+      console.error("Not supported: " + currentTab);
+      return prevState;
+    }
+  }
+
+  static formatDate(created_at) {
+    return moment(created_at).format("YYYY-MM-DD");
+  }
+
+  static processSamples(newSamples, genomes) {
+    const samples = newSamples.map(sample => {
+      const genome = genomes[sample.host_genome_id];
+      // TODO (gdingle): best description for blanks?
+      sample.host_genome = genome ? genome.name : "unknown";
+      sample.sample_tissue = sample.sample_tissue || "unknown";
```

Comment:
The `sample_tissue` column (and other similar "metadata" columns) in the sample model is deprecated. We are now using the `Metadatum` and `MetadataField` models to handle metadata more flexibly. There is still some old data in the sample_tissue column, but all new data is in the MetadataField `sample_type`. Would you mind working with @jshoe  / @MarkAZhang  to switch over to the new metadata models? Once that change has been made everywhere, we can drop the corresponding columns from the sample model. The duplication is confusing for new developers :)

Reply:
Got it. I'll clean this up in my next PR or a dedicated metadata migration PR.